File: component-service/src/main/java/dev/yidafu/pan/component/common/exception/ServiceException.java
package dev.yidafu.pan.component.common.exception;

import lombok.Data;

@Data
public class ServiceException extends RuntimeException {
    int code;

    String message;

    public ServiceException(String message) {
        super(message);
        this.message = message;
        code = 9999;
    }

    public ServiceException(int code, String message) {
        super(message);
        this.message = message;
        this.code = code;
    }

    public ServiceException(ExpectionCode code) {
        super(code.message);
        this.code = code.code;
        this.message = code.message;
    }
}

File: common/src/main/kotlin/dev/yidafu/pan/common/exception/ServiceException.kt
package dev.yidafu.pan.common.exception


open class ServiceException : RuntimeException {
    var code: Int
    override var message: String?

    constructor(message: String?) : super(message) {
        this.message = message
        code = 9999
    }

    constructor(code: Int, message: String?) : super(message) {
        this.message = message
        this.code = code
    }

    constructor(code: ExceptionCode) : super(code.message) {
        this.code = code.code
        message = code.message
    }
}